File: src/main/java/io/github/tanguygab/conditionalactions/actions/commands/GroupAction.java
package io.github.tanguygab.conditionalactions.actions.commands;

import io.github.tanguygab.conditionalactions.actions.Action;
import org.bukkit.OfflinePlayer;

public class GroupAction extends Action {
    public GroupAction() {
        super("(?i)group:( )?");
    }

    @Override
    public String getSuggestion() {
        return "group: <action group>";
    }

    @Override
    public void execute(OfflinePlayer player, String match) {
        match = parsePlaceholders(player,match);
        getPlugin().getActionManager().findAndExecute(player,match,true);
    }
}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/actions/types/commands/GroupAction.kt
package io.github.tanguygab.conditionalactions.actions.types.commands

import io.github.tanguygab.conditionalactions.actions.Action
import org.bukkit.OfflinePlayer

class GroupAction : Action("^(?i)group:( )?".toRegex()) {
    override fun getSuggestion() = "group: <action group>"

    override fun execute(player: OfflinePlayer?, match: String) {
        val match = parsePlaceholders(player, match)
        plugin.actionManager.findAndExecute(listOf(player), match, true)
    }
}
